src/main/java/org/jenkinsci/remoting/protocol/IOHub.java

```diff
@@ -83,6 +92,10 @@
      * Our selector.
      */
     private final Selector selector;
+    private volatile boolean winWatcherSelectorIsWaiting = false;
```

Comment:
You can also make this even simpler still... don't even use this boolean, just all `iohub.selector.wakeup()` every `SELECTOR_WAKEUP_TIMEOUT_MS` as long as `ioHubRunning` that way the thread becomes ultra simple